Shared/Samples/Group layers together/GroupLayersTogetherView.swift

```diff
@@ -0,0 +1,182 @@
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct GroupLayersTogetherView: View {
+    /// A scene with an imagery basemap and a world elevations source.
+    @State private var scene: ArcGIS.Scene = {
+        let scene = Scene(basemapStyle: .arcGISImagery)
+        
+        // Add base surface to the scene for elevation data.
+        let elevationSource = ArcGISTiledElevationSource(url: .worldElevationService)
+        let surface = Surface()
+        surface.addElevationSource(elevationSource)
+        scene.baseSurface = surface
+        return scene
+    }()
+    
+    /// The current viewpoint of the scene view.
+    @State private var viewpoint: Viewpoint?
+    
+    /// A Boolean value that indicates whether the layers sheet is showing.
+    @State private var isShowingLayersSheet = false
+    
+    var body: some View {
+        SceneView(scene: scene, viewpoint: viewpoint)
+            .onViewpointChanged(kind: .centerAndScale) { viewpoint = $0 }
+            .task {
+                // Add group layers to the scene as operational layers.
+                scene.addOperationalLayers([makeProjectAreaGroupLayer(), makeBuildingsGroupLayer()])
+                
+                // Ensure all group layers' childlayers are loaded.
+                for groupLayer in scene.operationalLayers as! [GroupLayer] {
+                    await groupLayer.layers.load()
+                }
+                
+                // Set the scene's viewpoint with the extent of the project area group layer.
+                if let extent = scene.operationalLayers.first?.fullExtent {
+                    let camera = Camera(lookingAt: extent.center, distance: 700, heading: 0, pitch: 60, roll: 0)
+                    viewpoint = Viewpoint(latitude: .nan, longitude: .nan, scale: .nan, camera: camera)
+                }
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    let layersButton = Button("Layers") {
+                        isShowingLayersSheet = true
+                    }
+                    
+                    if #available(iOS 16, *) {
+                        layersButton
+                            .popover(isPresented: $isShowingLayersSheet, arrowEdge: .bottom) {
+                                layersList
+                                    .presentationDetents([.fraction(0.5)])
+#if targetEnvironment(macCatalyst)
+                                    .frame(minWidth: 300, minHeight: 270)
+#else
+                                    .frame(minWidth: 320, minHeight: 390)
+#endif
+                            }
+                    } else {
+                        layersButton
+                            .sheet(isPresented: $isShowingLayersSheet, detents: [.medium]) {
+                                layersList
+                            }
+                    }
+                }
+            }
```

Comment:
Thank you for making this change. The logic should be refactored out of the toolbar.
```suggestion
     layersButton
```

```
@ViewBuilder private var layersButton: some View {
        let button = Button("Layers") {
            isShowingLayersSheet = true
        }
        
        if #available(iOS 16, *) {
            button
                .popover(isPresented: $isShowingLayersSheet, arrowEdge: .bottom) {
                    layersList
                        .presentationDetents([.fraction(0.5)])
#if targetEnvironment(macCatalyst)
                        .frame(minWidth: 300, minHeight: 270)
#else
                        .frame(minWidth: 320, minHeight: 390)
#endif
                }
        } else {
            button
                .sheet(isPresented: $isShowingLayersSheet, detents: [.medium]) {
                    layersList
                }
        }
    }
```